app/js/enhancedDate.js

```diff
@@ -1,5 +1,118 @@
-/* 
- * add your module definition for enhancedDate here
- * assign the module to a global variable
- * /
+var myDate = (function iife() {    
+    'use strict';
 
+    var theDate;
+    var now;
+    var dayNames = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 
+        'Friday', 'Saturday'];
+    var monthNames = ['Januray', 'February', 'March', 'April', 'May', 'June',
+        'July', 'August', 'September', 'November', 'December'];
+
+    // explicit object enhancedDate   
+    var enhancedDate = {    
+
+        checkDate: function checkDate(userDate) {
+            if (theDate instanceof Date) {
+                return true; 
+            } else {
+                return false;    
+            }
+        },
+
+        setDate: function setDate(userDate) { 
+            if (userDate instanceof Date) {
+                theDate = userDate;
```

Comment:
This works, but keep in mind that objects are passed by reference when passed to functions as arguments. What this means is if the object that was passed in is modified outside the module, it will modify the date (theDate) being stored internally at the same time. Instead, just instantiate a new object using the one passed in: `theDate = new Date(userDate);`